e3sm_diags/parameter/core_parameter.py

```diff
@@ -346,3 +356,12 @@ def _run_diag(self) -> List[Any]:
                     sys.exit()
 
         return results
+
+    def __setattr__(self, name: str, value: Any) -> None:
+        """Override setattr to ensure year attributes are padded when set."""
+        if name in YEAR_ATTRIBUTES and value not in [None, ""]:
+            # Validate and pad the year before setting the attribute
+            value = pad_year(value)
+
+        # Set the attribute using the superclass method
+        super().__setattr__(name, value)
```

Comment:
@chengzhuzhang This is an alternative solution to the YearProperty descriptor class in #973 for automating padding of year attributes when they are set, either through `__init__` or with `param.<year_attribute>`. 

As mentioned [here](https://github.com/E3SM-Project/e3sm_diags/pull/973#pullrequestreview-2829772955), I found that `e3sm_diags` runs into problem when copying attributes between parameter objects in `run.py` when we add use `YearProperty`. Generally, this should work but `e3sm_diags` has some complex ways of copying attributes that isn't straightforward to resolve without adding my code and complexity.

This version is straightforward and meets our needs.

Reply:
I think this is a good solution!